src/AbstractWalkers/AtomWalkers.jl

```diff
@@ -9,79 +9,187 @@ The `AtomWalker` struct represents a walker composed of atoms/molecules.
-    for walker in walkers
-        if walker.num_frozen_part > 0
-            e_frozen = frozen_energy(walker.configuration, lj, walker.num_frozen_part)
-            walker.energy_frozen_part = e_frozen
-        else
-            e_frozen = 0.0u"eV"
+function AtomWalker(configuration::FastSystem; freeze_species::Vector{Symbol}=Symbol[], merge_same_species=true)
+    list_num_par, configuration = sort_components_by_atomic_number(configuration, merge_same_species=merge_same_species)
+    C = length(list_num_par)
+    frozen = zeros(Bool, C)
+    if !isempty(freeze_species)
+        elements = unique(atomic_symbol(configuration))
+        @show elements
+        for species in freeze_species
+            if !(species in elements)
+                throw(ArgumentError("The species $species is not in the system."))
+            end
+        end
+        components = split_components(configuration, list_num_par)
+        for i in 1:length(freeze_species)
+            for j in 1:C
+                if freeze_species[i] in atomic_symbol(components[j])
+                    frozen[j] = true
+                end
+            end
         end
-        e_total = interaction_energy(walker.configuration, lj; frozen=walker.num_frozen_part) + e_frozen
-        walker.energy = e_total
     end
-    return walkers
+    return AtomWalker{C}(configuration, list_num_par=list_num_par, frozen=frozen)
+end
+
+
+"""
+    assign_energy!(walker::AtomWalker, lj::LennardJonesParametersSets)
+
+Assigns the energy to the given `walker` using the Lennard-Jones parameters `lj`.
+
+# Arguments
+- `walker::AtomWalker`: The walker object to assign the energy to.
+- `lj::LennardJonesParametersSets`: The Lennard-Jones parameters.
+
+# Returns
+- `walker::AtomWalker`: The walker object with the assigned energy.
+
+"""
+function assign_energy!(walker::AtomWalker, lj::LennardJonesParametersSets)
+    walker.energy_frozen_part = frozen_energy(walker.configuration, lj, walker.list_num_par, walker.frozen)
+    walker.energy = interacting_energy(walker.configuration, lj, walker.list_num_par, walker.frozen) + walker.energy_frozen_part
+    return walker
 end
 
+
 """
     struct LJAtomWalkers <: AtomWalkers
 
-The `LJAtomWalkers` struct contains a list of `AtomWalker` objects and the Lennard-Jones potential parameters 
-    that defines the interactions between the particles in the walkers.
+The `LJAtomWalkers` struct represents a collection of atom walkers that interact with each other using the Lennard-Jones potential.
 
 # Fields
-- `walkers::Vector{AtomWalker}`: The list of `AtomWalkers`.
-- `lj_potential::LJParameters`: The Lennard-Jones potential parameters.
+- `walkers::Vector{AtomWalker{C}}`: A vector of atom walkers, where `C` is the number of components.
+- `lj_potential::LennardJonesParametersSets`: The Lennard-Jones potential parameters. See `LennardJonesParametersSets`.
```

Comment:
It's kind of carried over from `LJParameters` and attached with `Sets`. I suggest we change it later, I sense it will be some changes around how we handle LJ potentials (or import it from another performant package).